### ISSUE TITLE
unlink problem while cleaning up the Smarty cache

### ISSUE THREAD

**Issue by dstuecken** [smarty-php/smarty#24]
We're experiencing E_WARNINGS with your unlink operation in smarty_internal_write_file.php:58 on PHP 5.6.3 on Windows, since it is not checking if the file exists.

An is_file wrapper may fixes this:

```
if (Smarty::$_IS_WINDOWS) {
            if (is_file($_filepath)) {
                // remove original file
                @unlink($_filepath);
            }
            // rename tmp file
            $success = @rename($_tmp_file, $_filepath);
        } else {
```

Full Stack trace:
```
E_WARNING: unlink(C:\xampp\htdocs\i-doit\src/themes/default/smarty/templates_c\de^ad3fc0693ee80efa44169c025761566a1d4777af.file.searchBar.tpl.php): No such file or directory
... 11 non-project frames
1
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_write_file.php" line 58 in unlink
2
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_write_file.php" line 58 in writeFile
3
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_template.php" line 200 in compileTemplateSource
4
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_templatebase.php" line 149 in fetch
5
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_template.php" line 284 in getSubTemplate
6
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_templatebase.php(151) : eval()'d code" line 40 in content_55361efcc69c05_24852998
7
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_templatebase.php" line 182 in fetch
8
File "C:\xampp\htdocs\i-doit\src\classes\libraries\smarty\libs\sysplugins\smarty_internal_templatebase.php" line 377 in display
9
File "C:\xampp\htdocs\i-doit\src\classes\components\isys_component_template.class.php" line 116 in display
10
File "C:\xampp\htdocs\i-doit\src\hypergate.inc.php" line 265 in include_once
11
File "C:\xampp\htdocs\i-doit\index.php" line 202 in <main>
```

**Comment by uwetews** [comment]
In the early days of Smarty 3 tests showed bad performance of code like this
if (file_exists($file)) {
  $foo = filemtime($file);
}
and other.

So  '@' error suppression was chosen for performance  reason and also deal with some rare race condition between concurrent PHP processes.

New tests showed that now is_file() is much faster.

dev-master is now updated to use is_file() checks but '@' error suppression is left to deal with the rare race conditions.

This change will be included in the soon upcoming 3.1.22 release.

**Comment by dstuecken** [comment]
Thanks a lot! :+1:  :-)
